src/modbus_rtu_client.cpp

```diff
@@ -2,14 +2,227 @@
+
+    DataVectorUint8 body = utils::build_read_holding_register_message_body(first_register_address,num_registers_to_read);
+
+    DataVectorUint8 full_message = full_message_from_body(body, consts::READ_HOLDING_REGISTER_MESSAGE_LENGTH, unit_id );
+
+    conn.send_bytes(full_message);
+    modbus::DataVectorUint8 response = conn.receive_bytes( max_adu_size() );
+    uint16_t payload_size = validate_response(response, full_message ) ;
+    return return_only_registers_bytes ? response_without_protocol_data( response , payload_size ) : response;
+
+    #else
+
+    static_assert( false , "implementaion done only for little endian" );
+
+    #endif
+
+}
+
+DataVectorUint8 ModbusDataContainerUint16::get_payload_as_bigendian() const {
+
+    DataVectorUint8 result;
+    result.reserve( m_payload.size() * sizeof( DataVectorUint16::value_type ));
+
+    // if ( m_byte_order == ByteOrder::BigEndian )
+    if ( m_byte_order == ByteOrder::LittleEndian )
+        for( uint16_t value : m_payload ) {
+            SwapIt s { value };
+            result.push_back( s.value_8.hi   );
+            result.push_back( s.value_8.low  );
+        }
+    else
+        for( uint16_t value : m_payload ) {
+            SwapIt s { value };
+            result.push_back( s.value_8.low );
+            result.push_back( s.value_8.hi  );
+        }
+
+    return result;
+}
+
+DataVectorUint8 everest::modbus::ModbusRTUClient::writer_multiple_registers( uint8_t unit_id,
+                                                                             uint16_t first_register_address,
+                                                                             uint16_t num_registers_to_write,
+                                                                             const ModbusDataContainerUint16& payload,
+                                                                             bool return_only_registers_bytes ) const {
+
+    #if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+
+    using namespace std::string_literals;
+
+    if ( num_registers_to_write  > everest::modbus::consts::rtu::MAX_REGISTER_PER_MESSAGE )
+        throw everest::modbus::exceptions::message_size_exception( ""s + __PRETTY_FUNCTION__ +
+                                                                   " Requested number of 16 bit registers " + std::to_string( num_registers_to_write ) +
+                                                                   " would exceed allowed message size of " + std::to_string( everest::modbus::consts::rtu::MAX_REGISTER_PER_MESSAGE ) + " register !" );
+
+    DataVectorUint8 body = utils::build_write_multiple_register_body(first_register_address, num_registers_to_write, payload );
+    DataVectorUint8 full_message = full_message_from_body(body, body.size() /* unused parameter */, unit_id );
```

Comment:
```suggestion
    DataVectorUint8 full_message = full_message_from_body( body, body.size() /* unused parameter */, unit_id );
```

Reply:
Also, if there is an `unused parameter`, can it be removed/replaced?

Reply:
a change here would break other inheriting classes.

Reply:
```suggestion
    DataVectorUint8 full_message = full_message_from_body( body, body.size() /* unused parameter, kept for inheritance compatibility */, unit_id );
```
How about this then?